lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp

```diff
@@ -5069,6 +5069,31 @@ void SwiftASTContext::PrintDiagnostics(DiagnosticManager &diagnostic_manager,
 
 void SwiftASTContext::ModulesDidLoad(ModuleList &module_list) {
   ClearModuleDependentCaches();
+  TargetSP target_sp = GetTarget().lock();
+  if (!target_sp)
+    return;
+  bool use_all_compiler_flags = target_sp->GetUseAllCompilerFlags();
+  unsigned num_images = module_list.GetSize();
+  for (size_t mi = 0; mi != num_images; ++mi) {
+    std::vector<std::string> module_search_paths;
+    std::vector<std::pair<std::string, bool>> framework_search_paths;
+    std::vector<std::string> extra_clang_args;
+    lldb::ModuleSP module_sp = module_list.GetModuleAtIndex(mi);
+
+    ProcessModule(module_sp, m_description, use_all_compiler_flags, *target_sp,
+                  module_search_paths, framework_search_paths,
+                  extra_clang_args);
+    if (use_all_compiler_flags && !extra_clang_args.empty()) {
```

Comment:
Is there anything you can add as a comment to about this condition? From this function body alone, I can read between some of the lines but not all.